packages/core/src/Table/stories/TableSection/TableFilter.tsx

```diff
@@ -0,0 +1,338 @@
+import { useMemo, useRef, useState } from "react";
+import { css, keyframes } from "@emotion/css";
+import {
+  HvButton,
+  HvCheckBox,
+  HvFilterGroup,
+  HvFilterGroupProps,
+  HvFilterGroupValue,
+  HvInput,
+  HvPagination,
+  HvTable,
+  HvTableBody,
+  HvTableCell,
+  HvTableColumnConfig,
+  HvTableContainer,
+  HvTableHead,
+  HvTableHeader,
+  HvTableRow,
+  HvTableSection,
+  HvTypography,
+  baseDropdownClasses,
+  theme,
+  useHvData,
+  useHvFilters,
+  useHvGlobalFilter,
+  useHvPagination,
+} from "@hitachivantara/uikit-react-core";
+import { Add, Close, Filters } from "@hitachivantara/uikit-react-icons";
+
+import { AssetEvent, getColumns, makeData, range } from "../storiesUtils";
+
+const EmptyRow = () => (
+  <HvTableRow>
+    <HvTableCell colSpan={100} />
+  </HvTableRow>
+);
+
+const slide = keyframes`
+  0% { 
+    max-height: 0;
+  }
+  100% { 
+    max-height: 300px;
+  }
+`;
+
+const classes = {
+  filtersContainer: css({
+    display: "flex",
+    width: "100%",
+    backgroundColor: theme.colors.warning_20,
+    border: `1px solid ${theme.colors.atmo3}`,
+    overflow: "hidden",
+    animation: `${slide} 1.5s ease-in-out`,
+  }),
+  filters: css({
+    display: "flex",
+    width: "100%",
+    padding: theme.space.xs,
+    justifyContent: "space-between",
+    alignItems: "center",
+  }),
+  gemsContainer: css({
+    display: "flex",
+    width: "100%",
+    justifyContent: "flex-start",
+    padding: `0px ${theme.space.xs}`,
+    flexWrap: "wrap",
+    gap: theme.space.xs,
+  }),
+  filterHeader: css({
+    borderColor: "transparent",
+    "&:hover": {
+      borderColor: "transparent",
+      backgroundColor: theme.colors.primary_20,
+    },
+    "&:focus-visible": {
+      borderColor: "transparent",
+    },
+    [`&.${baseDropdownClasses.headerOpen}`]: {
+      borderColor: theme.colors.secondary,
+      "&:hover": {
+        borderColor: theme.colors.secondary,
+      },
+    },
+  }),
+};
+
+const data = makeData(50);
+
+const filters: HvFilterGroupProps["filters"] = [
+  {
+    id: "status",
+    name: "Status",
+    data: [
+      { id: "closed", name: "Closed" },
+      { id: "open", name: "Open" },
+    ],
+  },
+  {
+    id: "severity",
+    name: "Severity",
+    data: [
+      {
+        id: "critical",
+        name: "Critical",
+      },
+      { id: "major", name: "Major" },
+      { id: "average", name: "Average" },
+      { id: "minor", name: "Minor" },
+    ],
+  },
+  {
+    id: "priority",
+    name: "Priority",
+    data: [
+      {
+        id: "high",
+        name: "High",
+      },
+      { id: "medium", name: "Medium" },
+      { id: "low", name: "Low" },
+    ],
+  },
+];
+
+export const TableFilter = () => {
+  const filterRef = useRef<HTMLDivElement>(null);
+
+  const [selectedFilters, setSelectedFilters] = useState<HvFilterGroupValue>();
+  const [checked, setChecked] = useState(false);
```

Comment:
I was half-expecting this to be turned on by default.
Using `HvFilterGroup` with an "AND" filtering logic is a bit strange to me, as selecting more than one checkbox (in a group) would always yield no results 🤔

Reply:
I used the AND behaviour has the default because that's the default react table behaviour when using `setAllFilters`. But I agree that it's a weird behaviour, that's why I added the OR one. I can change the sample and default to OR.